forest/cli/src/cli/send_cmd.rs

```diff
@@ -7,9 +7,69 @@ use forest_json::message::json::MessageJson;
 use forest_rpc_client::{mpool_push_message, wallet_default_address};
 use fvm_shared::{address::Address, econ::TokenAmount, message::Message, METHOD_SEND};
 use num::BigInt;
+use rust_decimal::prelude::*;
+use rust_decimal_macros::dec;
 
 use super::{handle_rpc_err, Config};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FILAmount {
+    pub value: TokenAmount,
+}
+
+impl FromStr for FILAmount {
+    type Err = anyhow::Error;
+
+    /// `FILAmount::from_str` parses a string (input to the 'amount' field of
+    /// the the `send` command) with all units supported by forest wallet
+    /// (`attoFIL`, `femtoFIL`, `picoFIL`, `nanoFIL`, `microFIL`,
+    /// `milliFIL`, and `FIL`) and converts the amount into an `attoFIL`
+    /// `TokenAmount`. To match the current behavior in Lotus, the default
+    /// units (if no units are specified in the command line) are `FIL`.
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let error_call = |e| anyhow::anyhow!("failed to parse fil amount: {}. {}.", &s, e);
```

Comment:
Is it better for `e` to be a `String` or `&str`?

Reply:
I phoned a friend, who told me it's better to be a`&str` to avoid the heap allocation (and as a bonus, this removes all the calls to `to_string()`). Just seems that `e` needs to be explicity typed in the closure or the borrow in the `parsed_val` call complains.